src/mediaembedediting.js

```diff
@@ -34,88 +34,108 @@ export default class MediaEmbedEditing extends Plugin {
 			providers: [
 				{
 					name: 'dailymotion',
-					url: /^(https:\/\/)?(www\.)?dailymotion\.com\/video\/(\w+)/,
-					html: id =>
-						'<div style="position: relative; padding-bottom: 100%; height: 0; ">' +
-							`<iframe src="https://www.dailymotion.com/embed/video/${ id }" ` +
-								'style="position: absolute; width: 100%; height: 100%; top: 0; left: 0;" ' +
-								'frameborder="0" width="480" height="270" allowfullscreen allow="autoplay">' +
-							'</iframe>' +
-						'</div>'
+					url: /^dailymotion\.com\/video\/(\w+)/,
+					html: match => {
+						const id = match.slice( -1 );
```

Comment:
You know exactly at which index of `match` this capturing group is available, so why complicating this with `.slice( -1 )`?